README.md

```diff
@@ -1188,11 +1188,45 @@ As mentioned before, the above code will compile. The `NewNullWriter` returns a
+What happened? An interface method in Go is essentially a function pointer. In this case, since we are pointing to the function of an interface, rather than an actual method implementation, we are trying to invoke a function that's actually a `nil` pointer. To avoid this from happening, we would have to provide the `NulllWriter` with a struct which fulfills the interface contract, with actual implemented methods.  
 
-> NOTE: Some people argue that using embedded interfaces is a good way of creating a mock structure for testing a subset of interface methods. Essentially, by using an embedded interface, you won't have to implement all of the methods of the interface; rather, you can choose to implement only the few methods that you'd like to test. Within the context of testing/mocking, I can see this argument, but I am still not a fan of this approach.
+```go
+func main() {
+  w := NullWriter{
+    Writer: &bytes.Buffer{},
+  }
+
+    w.Write([]byte{1, 2, 3})
+}
+```
+
+> NOTE: In the above examble, `Writer` is referring to the embedded `io.Writer` interface. It is also possible to invoke the `Write` method, by accessing this property with: `w.Writer.Write()`
+
+We are no longer receiving a panic and can now use the `NullWriter` as a `Writer`. This initialisation process is not much different from having properties which are initialised as `nil`, as discussed previously. Therefore, logically, we should try and handle them in a similar way. However, this is where embedded interfaces become a little difficult to work with. In a previous chapter, it was explained that the best way to handle potential `nil` values, was to make the property in question private and create a public *getter* method. This way, we could ensure that our property is, in fact, not `nil`. Unfortunately, this is simply not possible with embedded interfaces, as they are by nature, always public. 
+
+Another concern raised by using embedded interfaces, is the potential confusion caused by partially overwritten interface methods: 
+
+```go
+type MyReadCloser struct {
+  io.ReadCloser
+}
+
+func (closer *ReadCloser) Read(data []byte) { ... }
+
+func main() {
+  closer := MyReadCloser{}
+  
+  closer.Read([]byte{1, 2, 3}) 	// works fine
+  closer.Close() 								// causes panic
+  closer.ReadCloser.Closer() 		// no panic 
+}
+```
+
+Even though, this might look like overriding methods, which are common in languages such as C# and Java. It isn't. Go doesn't have inheritance nor super classes. We can imitate the behaviour, but it is not an in-built part of the language. By using methods such as interface embedding without caution, we are creating confusing and possibly buggy code, just to save a few more lines of code. 
```

Comment:
```suggestion
Even though this might look like we're overriding methods, which is common in languages such as C# and Java, we actually aren't. Go has neither inheritance nor superclasses. We can imitate the behaviour, but it is not a built-in part of the language. By using methods such as interface embedding without caution, we can create confusing and potentially buggy code, just to save a few more lines.
```